hack/multikueue-e2e-test.sh

```diff
@@ -20,7 +20,6 @@ set -o pipefail
 
 SOURCE_DIR="$(cd "$(dirname -- "${BASH_SOURCE[0]}")" && pwd -P)"
 ROOT_DIR="$SOURCE_DIR/.."
-export E2E_TEST_IMAGE=gcr.io/k8s-staging-perf-tests/sleep:v0.1.0
```

Comment:
And here also. On this way try to use my previous suggestion https://github.com/kubernetes-sigs/kueue/pull/2570#discussion_r1744929902. It should work for both cases.

Reply:
Hey @mbobrovskyi I have updated tests accordingly but not sure why PR check e2e tests are failing :thinking:

Reply:
Yes, this is a good question. I can see an error `Error: failed to get image from containerd "sha256:d483aa6092983e8adea8d92d5661f7191f2e1b07ac480ec3a60a1584fea39a3c": image "docker.io/library/import-2024-09-13@sha256:12abe35a0f1d3da31f825e36ed20a1978ca1848d9973aa6c224fd5725ac6ebbc": not found`.

```
kubectl describe pods test-job-1-ckbj8 -n e2e-9prhl
Name:                 test-job-1-ckbj8
Namespace:            e2e-9prhl
Priority:             100
Priority Class Name:  high
Service Account:      default
Node:                 kind-worker2/172.18.0.2
Start Time:           Fri, 13 Sep 2024 09:02:24 +0300
Labels:               batch.kubernetes.io/controller-uid=3d540ff4-b19d-4aad-9b3f-4cf0af1f0ec6
                      batch.kubernetes.io/job-name=test-job-1
                      controller-uid=3d540ff4-b19d-4aad-9b3f-4cf0af1f0ec6
                      job-name=test-job-1
Annotations:          <none>
Status:               Pending
IP:                   10.244.2.10
IPs:
  IP:           10.244.2.10
Controlled By:  Job/test-job-1
Containers:
  c:
    Container ID:  
    Image:         gcr.io/k8s-staging-perf-tests/sleep:v0.1.0@sha256:8d91ddf9f145b66475efda1a1b52269be542292891b5de2a7fad944052bab6ea
    Image ID:      
    Port:          <none>
    Host Port:     <none>
    Args:
      1m
    State:          Waiting
      Reason:       CreateContainerError
    Ready:          False
    Restart Count:  0
    Requests:
      cpu:        1
    Environment:  <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-f6424 (ro)
Conditions:
  Type                        Status
  PodReadyToStartContainers   True 
  Initialized                 True 
  Ready                       False 
  ContainersReady             False 
  PodScheduled                True 
Volumes:
  kube-api-access-f6424:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age               From               Message
  ----     ------     ----              ----               -------
  Normal   Scheduled  27s               default-scheduler  Successfully assigned e2e-9prhl/test-job-1-ckbj8 to kind-worker2
  Normal   Pulled     3s (x4 over 27s)  kubelet            Container image "gcr.io/k8s-staging-perf-tests/sleep:v0.1.0@sha256:8d91ddf9f145b66475efda1a1b52269be542292891b5de2a7fad944052bab6ea" already present on machine
  Warning  Failed     3s (x4 over 27s)  kubelet            Error: failed to get image from containerd "sha256:d483aa6092983e8adea8d92d5661f7191f2e1b07ac480ec3a60a1584fea39a3c": image "docker.io/library/import-2024-09-13@sha256:12abe35a0f1d3da31f825e36ed20a1978ca1848d9973aa6c224fd5725ac6ebbc": not found
```

Reply:
As I can see [here](https://kubernetes.io/docs/concepts/containers/images/#image-names): `Image name with tag and digest. Only digest will be used for pulling.` So no difference.

Reply:
So even if we keep image with tag+digest it will work, then wondering why the PR checks are failing :eyes:

Reply:
Yes, it should work. I think the problem on kind.

Reply:
Same error scenario is mentioned here : https://github.com/kubernetes-sigs/kind/issues/2922#issuecomment-1247572299

Reply:
And also here: https://github.com/kubernetes-sigs/kind/issues/2394.

Reply:
If I correct understood we can load image to kind by digest but we can't reference to it. So we still need tag to do it.

Reply:
We can also create tag manually for previous pulled image by digest and load it by `kind load docker-image gcr.io/k8s-staging-perf-tests/sleep:manual-created-tag`. I tested it locally, this solution working well, but I think it's very complicated. And also I'm not sure that this solution will cover current problematic of this PR.

Reply:
> We can also create tag manually for previous pulled image by digest and load it by kind load docker-image gcr.io/k8s-staging-perf-tests/sleep:manual-created-tag. I tested it locally, this solution working well, but I think it's very complicated. And also I'm not sure that this solution will cover problematic of this PR.

We discussed it with @mimowo and found it doesn't sound too bad.

So I propose to make it like this:

```go
const (
	// E2eTestSleepImage is the image used for testing.
	E2eTestSleepImage = "gcr.io/k8s-staging-perf-tests/sleep:v0.1.0@sha256:8d91ddf9f145b66475efda1a1b52269be542292891b5de2a7fad944052bab6ea"
)
```

```shell
# sleep image to use for testing.
export E2E_TEST_SLEEP_IMAGE=gcr.io/k8s-staging-perf-tests/sleep:v0.1.0@sha256:8d91ddf9f145b66475efda1a1b52269be542292891b5de2a7fad944052bab6ea
```

```shell
# $1 cluster
function cluster_kind_load {
	sleep_image_name=${E2E_TEST_SLEEP_IMAGE%%:*}
	sleep_image_tag_and_digest="${E2E_TEST_SLEEP_IMAGE#*:}"
	sleep_image_tag="${sleep_image_tag_and_digest%@*}"

	# When we are using image name with tag and digest only digest will be used for pulling.
	# We can load image by a digest but we cannot reference it by the digest that we pulled.
	# This is because image digests hash the entire image including the metadata
	# that contains the repo tags in the image, and they will not match.
	# For more information https://github.com/kubernetes-sigs/kind/issues/2394#issuecomment-888713831.
	# But we can manually create tag for image with digest that we already pulled.
	docker tag $E2E_TEST_SLEEP_IMAGE "$sleep_image_name:$sleep_image_tag"

	cluster_kind_load_image "$1" "${E2E_TEST_SLEEP_IMAGE%@*}"
	cluster_kind_load_image "$1" "$IMAGE_TAG"
}
```

Reply:
I also tested this solution locally with removing images, so it should work fine.

Reply:
I like the proposal, just a small question:

```
# $1 cluster
function cluster_kind_load {
	e2e_test_sleep_image_without_sha=${E2E_TEST_SLEEP_IMAGE%%@*}

	# ...
	docker tag $E2E_TEST_SLEEP_IMAGE "$e2e_test_sleep_image_without_sha"

	cluster_kind_load_image "$1" "${e2e_test_sleep_image_without_sha}"
	cluster_kind_load_image "$1" "$IMAGE_TAG"
}
```
would it work @mbobrovskyi ? Seems simpler as there are fewer steps.

Reply:
Yeap, it works.

Reply:
@abhijeet-dhumal could you please apply this changes?

Reply:
Thank you both @mbobrovskyi @mimowo !
I truely appreciate your dedicated efforts toward enhancing this PR. :blush: